sdjournal/journal.go

```diff
@@ -357,7 +357,7 @@ func NewJournalFromDir(path string) (j *Journal, err error) {
 		return nil, err
 	}
 	defer func() {
-		if err == nil {
+		if err != nil {
```

Comment:
I think this was meant to be a resource cleaner on error, and this is now subverting the logic. OTOH this snippet looks over-complex, and I think the root issue here is implicitly reusing `err` for multiple purposes.